WordPress/Classes/System/ViewControllerNavigationAction.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+/// `ViewControllerNavigationAction` encapsulates a navigation action performed on a view controller.
+/// It includes the capability to chain these actions together to perform a sequence of navigations.
+/// This sequence is internally represented as a linked list where each node represents a navigation action.
+/// The `then` method is used to chain navigation actions, and the `start` method initiates the sequence.
+final class ViewControllerNavigationAction {
```

Comment:
It looks nice but I am bit concerned about adding one-off tools as such that have a different style of use but isn't widely used. Due to our lack of conventions in the codebase, we have plenty of such styles that are only used once. I think when we have this kind of tools in abundance, it pollutes the codebase due to the fact that one has to spend time understanding the functionality each time he/she faces a different one. 

This is not to say that we shouldn't use this or any particular tool. I only wanted to highlight that maybe we would want to spread the information and possibly make a minor architectural tool out of that we often use to solve specific problems. Just like Design Patterns having names and specific uses. 

Otherwise we have many different overlapping tools solving similar tasks that have different names.

I would still favor keeping this. But I'm curious if you have felt the same at all?

Reply:
> I would still favor keeping this. But I'm curious if you have felt the same at all?

💯 agree, and I have felt the same indeed. At first, this class was actually private in the `RootViewPresenter+AppSettings` file, because my intention wasn't to turn this class into some sort of library to be used everywhere.

Then I turned it into a public class in case we add more navigation methods in `RootViewPresenter`.

So instead of having that class in a global namespace like it is right now, what do you say we embed it in inside `RootViewPresenter` like this:

```swift
extension RootViewPresenter {
   final class ViewControllerNavigationAction {
         // ...
   }
}
```

Reply:
I also want to mention that adding yet another tool was the last option, I considered the following options first:

1. Chaining those navigations using callbacks. Like this:
```swift
navigateToMe() {
   navigateToAppSettings() {
        navigateToPrivacySettings() {
            // ... 
       }
   }
}
```
But you can see that this callback hell is not scalable. 

2. I thought of using async / await
```
func navigateToPrivacy() async {
await navigateToMe()
await navigateToAppSettings()
await navigateToPrivacySettings()
}
```
This didn't work either because the async / await makes the code be executed in a background thread, but this navigation should happen in the main thread.

_N.B: The above code is just a pseudo-code to convey the idea._

Reply:
It makes sense. We could do the extension to keep it lowkey for now 👍 . 

Just as a reference, for the main thread bit, I think @MainActor could be what you are looking for there.